feat: Documentation about ok-to-test label (#4345)
kubernetes/community: contributors/guide/pull-requests.md

That's the last step. Your pull request is now merged.

## More About `Ok-To-Test`

- For a Contributor, A `ok-to-test` label may just mean regular CI Testing to be done.
- For the reviewer or the member, labelling the PR with `ok-to-test` it means a lot more:

>     + They need to take care if the PR is not a wastage of our `CI/CD` resources.

Author: Sure making the updates

    + Is the PR worth testing or does it needs more change before going through the `CI/CD` process ?
    + Is the PR getting used to run malicious code to misuse our resources ?

> - An `ok-to-test` label may reduce the workload and smoothens the contributors experience as they can know if there is any failing test. If there is, you can fix the test and they don't have to wait for a long time to get a review from `maintainer/assignee`.

Reviewer: I would actually remove this line or put a brief note at the top along the lines of:
The `ok-to-test` label is applied by org members to PRs from external contributors. It signals that the PR can be tested.

- There are various other factors on which labelling of `ok-to-test` depends :
    + Size of PR :

>         * If the PR is of `size/S` or `size/M` which is just to fix a flake or test or spelling mistake, the reviewer can trigger the `CI/CD` without having a second thought.

Reviewer: I'm hesitant on this one, it's usually okay for things like spelling, grammer, or link fixes -- what we refer as [trivial edits](https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#10-trivial-edits).

Tests, flakes etc should still be evaluated.

        * If the PR is of `size/XXL` which aims at adding a new feature, a new API endpoint or any new substantial feature. There needs to other conventions & process to be followed regarding the change made. Hence it may have a slight to delay to get labelled with `ok-to-test`.

>     + Other maintainers or contributor who are not assigned to the following PR may also label `ok-to-test` , if the change is small.

Reviewer: ```suggestion
    + Other org members who are assigned to the PR may also label `ok-to-test`, if the change is small.
```

We try to avoid using the maintainer keyword and opt for member/reviewer/approver or contributor.

    + If the PR is labelled with `cncf-cla: no`, then it is better to wait before labelling `ok-to-test`.
    + PRs with tag `do-not-merge/hold` or `needs-rebase` should make the appropriate changes before the PR can be labelled `ok-to-test`.

>     + PRs created by mistake without a meaningful change of code should not also get labelled `ok-to-test`.

Reviewer: ```suggestion
    + PRs created by mistake without a meaningful change to code should not be labeled with `ok-to-test` and closed.
```
Mistakes should just be closed 👍 or ping the author about it and close.
